services/explorer-indexer/src/modules/node/historicalSyncer.service.ts

```diff
@@ -163,114 +174,227 @@ export default class HistoricalSyncService {
+        blockToSave.transactionCount = block.transactions?.length;
+
+        const savedBlock = await this.pbftService.safeSavePbft(blockToSave);
+
+        if (savedBlock) {
+          this.logger.log(`Finalized block ${savedBlock.hash}`);
         }
-      }
-      const fetchTxEnd = new Date();
-      console.log(
-        `Fetched ${block.transactions?.length} tx-es in ${
-          fetchTxEnd.getTime() - fetchTxStart.getTime()
-        } ms.`
-      );
-
-      if (!block.hash) {
-        continue;
-      }
 
-      let blockReward = new BigInteger('0', 10);
+        // update sync state
+        this.syncState.number = block.number;
+        this.syncState.genesis = this.chainState.genesis;
+        this.syncState.hash = block.hash;
 
-      // @note : Right now there is no way to get the genesis transactions to set:
-      // Initial validators, initial balances for delegators and faucet.
-      // big TODO
-
-      const txReceipts = await this.getTransactionReceipts(
-        blockTxs.map((tx) => tx.hash),
-        20
-      );
-
-      Array.from(new Set(blockTxs)).forEach((tx: ITransaction, idx: number) => {
-        const receipt = txReceipts[idx];
-        tx.status = parseInt(receipt.status, 16);
-        tx.index = parseInt(receipt.transactionIndex, 16);
-        const gasUsed = new BigInteger(receipt.gasUsed || '0x0', 16);
-        tx.gasUsed = gasUsed.toString();
-        const gasPrice = new BigInteger(tx.gasPrice || '0x0', 16);
-        tx.gasPrice = gasPrice.toString();
-        tx.cumulativeGasUsed = Number(
-          new BigInteger(receipt.cumulativeGasUsed || '0x0', 16)
+        const completed = new Date();
+        console.log(
+          'Synced block',
+          this.syncState.number,
+          this.syncState.hash,
+          'with',
+          block.transactions?.length,
+          'transactions, in',
+          completed.getTime() - started.getTime(),
+          'ms'
         );
-        const reward = gasUsed.multiply(gasPrice);
-        const newVal = blockReward.add(reward);
-        blockReward = newVal;
-      });
-
-      block.reward = blockReward.toString();
-      const blockToSave = { ...block };
-      blockToSave.transactions = blockTxs;
-      blockToSave.transactionCount = blockTxs.length;
-
-      const savedBlock = await this.pbftService.safeSavePbft(blockToSave);
-
-      if (savedBlock) {
-        this.logger.log(`Finalized block ${savedBlock.hash}`);
       }
-
-      // update sync state
-      this.syncState.number = block.number;
-      this.syncState.genesis = this.chainState.genesis;
-      this.syncState.hash = block.hash;
-
-      const completed = new Date();
-      console.log(
-        'Synced block',
-        this.syncState.number,
-        this.syncState.hash,
-        'with',
-        block.transactions?.length,
-        'transactions, in',
-        completed.getTime() - started.getTime(),
-        'ms'
-      );
     }
+
+    // sync blocks to tip
```

Comment:
Add a comment above to remember to remove this after we confirm the GraphQl implementation